homeassistant/components/co2signal/config_flow.py

```diff
@@ -117,19 +118,27 @@ async def _validate_and_create(
         """Validate data and show form if it is invalid."""
         errors: dict[str, str] = {}
 
-        try:
-            await self.hass.async_add_executor_job(get_data, self.hass, data)
-        except InvalidAuth:
-            errors["base"] = "invalid_auth"
-        except APIRatelimitExceeded:
-            errors["base"] = "api_ratelimit"
-        except Exception:  # pylint: disable=broad-except
-            errors["base"] = "unknown"
-        else:
-            return self.async_create_entry(
-                title=get_extra_name(data) or "CO2 Signal",
-                data=data,
-            )
+        session = async_get_clientsession(self.hass)
+        async with ElectricityMaps(token=data[CONF_API_KEY], session=session) as em:
+            try:
+                if CONF_COUNTRY_CODE in data:
+                    await em.latest_carbon_intensity_by_country_code(
+                        code=data[CONF_COUNTRY_CODE]
+                    )
+                else:
+                    await em.latest_carbon_intensity_by_coordinates(
+                        lat=data.get(CONF_LATITUDE, self.hass.config.latitude),
+                        lon=data.get(CONF_LONGITUDE, self.hass.config.longitude),
+                    )
```

Comment:
This looks similar to 
https://github.com/home-assistant/core/pull/101955/files#diff-e751aa6f65cfe751a820fef40f11ba70c0047184a78641055790590a5b4c17d3R44

I think you could extract the inner block into a helper function to make it more dry

Reply:
Addressed, moved to a helper function